File: ios/telobike/telobike/TBSearchResultTableViewCell.m
//
//  TBSearchResultTableViewCell.m
//  telobike
//
//  Created by Elad Ben-Israel on 1/27/14.
//  Copyright (c) 2014 Elad Ben-Israel. All rights reserved.
//

#import "TBSearchResultTableViewCell.h"

@interface TBSearchResultTableViewCell ()

@property (strong, nonatomic) IBOutlet UIImageView* iconView;
@property (strong, nonatomic) IBOutlet UILabel* resultTitleLabel;
@property (strong, nonatomic) IBOutlet UILabel* distanceLabel;

@end

@implementation TBSearchResultTableViewCell

- (void)prepareForReuse {
    [super prepareForReuse];
    self.iconView.image = nil;
    self.resultTitleLabel.text = nil;
    self.distanceLabel.text = nil;
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];
}

- (void)setImage:(UIImage *)image {
    self.iconView.image = image;
}

- (void)setTitle:(NSString *)title {
    self.resultTitleLabel.text = title;
}

- (void)setDetail:(NSString *)detail {
    self.distanceLabel.text = detail;
}

@end

File: ios/telobike/telobike/TBSearchResultTableViewCell.swift
//
//  TBSearchResultTableViewCell.swift
//  telobike
//
//  Created by Elad Ben-Israel on 11/16/14.
//  Copyright (c) 2014 Elad Ben-Israel. All rights reserved.
//

import Foundation

class TBSearchResultTableViewCell: UITableViewCell {
    
    @IBOutlet private weak var iconView: UIImageView!
    @IBOutlet private weak var resultTitleLabel: UILabel!
    @IBOutlet private weak var distanceLabel: UILabel!
    
    var icon: UIImage? {
        didSet {
            self.iconView.image = icon
        }
    }

    var title: String? {
        didSet {
            self.resultTitleLabel.text = self.title
        }
    }
    
    var detail: String? {
        didSet {
            self.distanceLabel.text = self.detail
        }
    }
    
    override func prepareForReuse() {
        super.prepareForReuse()
        self.iconView.image = nil
        self.resultTitleLabel.text = nil
        self.distanceLabel.text = nil
    }
}
